### PR TITLE
Provisional alert

### PR DESCRIPTION
**Pull request by Borges-Porteous**
First step:(:) see if it compiles!

### CODE REVIEW

File: src/_experimental.scss [hunk 1]
```diff
@@ -28,14 +28,14 @@
 		}
 	}
 
-	// PinkDay color
-	.bg-pnkDy {
-		background-color: #AD005F;
+	// // PinkDay color NA
+	// .bg-pnkDy {
```
Comment (duboisp) [inline src/_experimental.scss]: This should not be in comment, If there a removal it should be submited in a different pull request

File: src/_experimental.scss [hunk 2]
```diff
@@ -28,14 +28,14 @@
...
+	// 	background-color: #AD005F;
 
-		&.well.header-rwd {
-			background-color: #570030;
-		}
-	}
+	// 	&.well.header-rwd {
+	// 		background-color: #570030;
```
Comment (duboisp) [inline src/_experimental.scss]: This should not be in comment, If there a removal it should be submited in a different pull request

File: src/_experimental.scss [hunk 3]
```diff
@@ -159,3 +159,138 @@
...
+			color: #d3080c;
+			content: "\e101";
+		}
+
+		&.alert-bg {
+			background-color: #f3e9e8;
+
+			details > summary {
+				color: #843534;
+				font-weight: 700;
```
Comment (duboisp) [inline src/_experimental.scss]: It could be preferable to use mixins instead of copying the same rule. 

Here an example:
How to define a mixin https://github.com/wet-boew/wet-boew/blob/master/src/plugins/multimedia/_base.scss#L5 
How to use it: https://github.com/wet-boew/wet-boew/blob/master/src/plugins/multimedia/_base.scss#L104

Comment (Borges-Porteous) [reply src/_experimental.scss]: Should I put the mixin variables at the top of the experimental.scss file? Or above the alerts section in the experimental.scss file? Or in the variables.scss/somewhere else in GCWeb?

Comment (Borges-Porteous) [reply src/_experimental.scss]: @duboisp I haven't finished this - would be great if you could advise where to put them and I'll update this tomorrow (Thursday)

Comment (Borges-Porteous) [reply src/_experimental.scss]: I've gone ahead and defined them at the top of the _experimental.scss sheet. We can move them elsewhere if needed...

Should be ready now.

Comment (duboisp) [reply src/_experimental.scss]: The mixin is only used by SASS during the compilation into CSS. So yes it is correct at the top of the file.

File: src/_experimental.scss [hunk 4]
```diff
@@ -159,3 +169,102 @@
 		}
 	}
 }
+
+
+.provisional {
+	.alert-info > :first-child::before,
```
Comment (duboisp) [inline src/_experimental.scss]: I think this block is to undo the old style. Can you add a comment before that said that?

File: src/_experimental.scss [hunk 5]
```diff
@@ -4,6 +4,16 @@
 	- Experimental  CSSspecific for view port are defined in the "views/[view port]" theme CSS
 */
 
+// $alert-green--color: #2b542c;
+$danger-color: #d3080c;
```
Comment (duboisp) [inline src/_experimental.scss]: Add a comment before to mention those new color is for the provisional alert.
For each those new SASS variable, can you prefix them with "provisional" ? It is just to avoid possible conflict with existing code.

File: src/_experimental.scss [hunk 6]
```diff
@@ -4,6 +4,16 @@
...
+$warning-color: #f90;
+$info-color: #269abc;
+$success-color: #278400;
+
+%alert-details-summary {
+	font-weight: 700;
+}
+
 .experimental {
```
Comment (duboisp) [inline src/_experimental.scss]: Do you mind to update this line for:
```
.experimental, .provisional {
```

File: src/_experimental.scss [hunk 7]
```diff
@@ -159,3 +169,102 @@
 		}
 	}
 }
+
+
+.provisional {
```
Comment (duboisp) [inline src/_experimental.scss]: This is perfect, keep it as is

👍

File: src/_experimental.scss [hunk 8]
```diff
@@ -159,3 +169,102 @@
...
+	.alert-info > :first-child::before,
+	.alert-success > :first-child::before,
+	.alert-warning > :first-child::before,
+	.alert-danger > :first-child::before {
+		content: none;
+	}
+
+	.alert {
```
Comment (duboisp) [inline src/_experimental.scss]: Along with the markup change proposed, you can use the parent selector ```&:```

fyi ref. https://sass-lang.com/documentation/style-rules/parent-selector

Comment (Borges-Porteous) [reply src/_experimental.scss]: Not sure how to action this one. 

Did you mean on the .alert-info > :first-child::before, .alert-success > :first-child::before, etc?

Should that be .alert-info, .alert-success, .alert-warning, .alert-danger { &> :first-child::before { ?

File: site/pages/experimental-en.hbs [hunk 9]
```diff
@@ -243,3 +256,46 @@ if ( s.getTime() &lt; msT && msT &lt; e.getTime() ) {
 		</div>
 	</div>
 </div>
+
+<h3 id="beta-alert">Beta alert designs</h3>
+<div class="provisional">
+	<details class="alert alert-danger">
```
Comment (duboisp) [inline site/pages/experimental-en.hbs]: provisional css class name need to be sibling CSS of alert and not be the parent

File: site/pages/experimental-fr.hbs [hunk 10]
```diff
@@ -242,3 +255,42 @@ if ( s.getTime() &lt; msT && msT &lt; e.getTime() ) {
 		</div>
 	</div>
 </div>
+
+<h3 id="beta-alert">Beta alert designs</h3>
```
Comment (delisma) [inline site/pages/experimental-fr.hbs]: @Borges-Porteous Can we remove the beta mention? Since it's already in a provisional process

File: src/_experimental.scss (hunk 8 above)
Comment (duboisp) [reply src/_experimental.scss]: The code will look like this
```
<div class="provisional alert alert-warning">
	<h2>Alerte d’avertissement</h2>
	<p>Détails optionnel d'alerte.</p>
</div>
```

And yes add it to the other CSS class name, like "alert-warning, ..."

File: src/_experimental.scss [hunk 11]
```diff
@@ -159,3 +165,103 @@
...
+
+		&::before {
+			color: $provisional-warning-color;
+			content: "\e107";
+		}
+	}
+
+	.alert-danger {
+		border-color: $provisional-danger-color;
+		border-image: linear-gradient(to bottom, $provisional-danger-color 16px, $provisional-danger-color 16px, rgba(0, 0, 0, 0) 16px, rgba(0, 0, 0, 0) 48px, $provisional-danger-color 48px, $provisional-danger-color 48px) 1 100%;
```
Comment (duboisp) [inline src/_experimental.scss]: This would be a nice to have, it will be to try to remove CSS code duplication.

SASS already have a few feature to help to avoid code duplication like
* `@ function` ref.https://sass-lang.com/documentation/at-rules/function
* The share plugin has something similar too. ref. https://github.com/wet-boew/wet-boew/blob/master/src/plugins/share/_base.scss#L7

File: src/_experimental.scss (hunk 8 above)
Comment (EricDunsworth) [reply src/_experimental.scss]: Addressed in #1650.

File: site/pages/experimental-fr.hbs (hunk 10 above)
Comment (EricDunsworth) [reply site/pages/experimental-fr.hbs]: Addressed in #1650. I removed all mentions of "beta" with regards to the alerts.

File: src/_experimental.scss (hunk 11 above)
Comment (EricDunsworth) [reply src/_experimental.scss]: Addressed in #1650. Went with a mixin.

**Comment by duboisp** [comment]
Provisional feature added via #1661
